### PR TITLE
Change dryrun to not include format args if dry run is false

### PR DESCRIPTION
**Pull request by billonahill**
I noticed when testing that dry run format args were being passed with empty values even when dry-run was not enabled. Change to not pass format if dry run is not passed.

Also SubmitterMain was using the wrong option char for dry run format.

### CODE REVIEW

File: heron/spi/src/java/com/twitter/heron/spi/utils/ReflectionUtils.java [hunk 1]
```diff
@@ -28,6 +28,9 @@ private ReflectionUtils() {
   @SuppressWarnings("unchecked") // we don't know what T is until runtime
   public static <T> T newInstance(ClassLoader classLoader, String className)
       throws ClassNotFoundException, IllegalAccessException, InstantiationException {
+    if (className == null) {
+      throw new ClassNotFoundException("Can not instantiate class. className must not be null");
+    }
     return (T) classLoader.loadClass(className).newInstance();
```
Comment (huijunw) [inline heron/spi/src/java/com/twitter/heron/spi/utils/ReflectionUtils.java]: loadClass() throws ClassNotFoundException. why do we throw it manually?

Comment (objmagic) [inline heron/spi/src/java/com/twitter/heron/spi/utils/ReflectionUtils.java]: Thanks. I didn't review carefully here. @billonahill do you have any particular reason to add this check?

Comment (billonahill) [inline heron/spi/src/java/com/twitter/heron/spi/utils/ReflectionUtils.java]: loadClass throws NullPointerException when passed null. The calling classes catch ClassNotFoundException but not NPE.

**Comment by objmagic** [comment]
LGTM.

**Pull request merged**